adapters/repos/db/lsmkv/segment.go

```diff
@@ -69,6 +69,7 @@ func newSegment(path string, logger logrus.FieldLogger, metrics *Metrics,
 	if err != nil {
 		return nil, errors.Wrap(err, "open file")
 	}
+	defer file.Close()
```

Comment:
Thanks! I'm surprised that this one doesn't cause any issues. The contents of the file are `mmaped` in the next couple of liens, I would have expected that the file needs to stay open while the memory mapping is active. But all tests are still green, so I guess it does not.